src/internal/m365/collection/drive/delta_tree_test.go

```diff
@@ -1205,3 +1205,231 @@ func (suite *DeltaTreeUnitSuite) TestFolderyMcFolderFace_GenerateCollectables()
+				folderID("c3"): d.strPath(t, folderName("o/ld"), folderName("c2"), folderName("c3")),
+			},
+			expect: map[string]string{
+				rootID:         d.strPath(t),
+				folderID():     d.strPath(t, folderName("n/ew")),
+				folderID("c1"): d.strPath(t, folderName("n/ew"), folderName("c1")),
+				folderID("c2"): d.strPath(t, folderName("n/ew"), folderName("c2")),
+				folderID("c3"): d.strPath(t, folderName("n/ew"), folderName("c2"), folderName("c3")),
+			},
+		},
+		{
+			name: "tombstoned directories get removed",
+			collectables: map[string]collectable{
+				rootID:     {currPath: d.fullPath(t)},
+				folderID(): {prevPath: d.fullPath(t, folderName("old"))},
+			},
+			prevPaths: map[string]string{
+				rootID:         d.strPath(t),
+				folderID():     d.strPath(t, folderName("old")),
+				folderID("c1"): d.strPath(t, folderName("old"), folderName("c1")),
+				folderID("c2"): d.strPath(t, folderName("old"), folderName("c2")),
+				folderID("c3"): d.strPath(t, folderName("old"), folderName("c2"), folderName("c3")),
+			},
+			expect: map[string]string{
+				rootID: d.strPath(t),
+			},
+		},
+		{
+			name: "mix of moved and tombstoned",
+			collectables: map[string]collectable{
+				rootID: {currPath: d.fullPath(t)},
+				folderID(): {
+					prevPath: d.fullPath(t, folderName("old")),
+					currPath: d.fullPath(t, folderName()),
+				},
+				folderID("c3"): {prevPath: d.fullPath(t, folderName("old"), folderName("c2"), folderName("c3"))},
+			},
+			prevPaths: map[string]string{
+				rootID:         d.strPath(t),
+				folderID():     d.strPath(t, folderName("old")),
+				folderID("c1"): d.strPath(t, folderName("old"), folderName("c1")),
+				folderID("c2"): d.strPath(t, folderName("old"), folderName("c2")),
+				folderID("c3"): d.strPath(t, folderName("old"), folderName("c2"), folderName("c3")),
+				folderID("c4"): d.strPath(t, folderName("old"), folderName("c2"), folderName("c3"), folderName("c4")),
+			},
+			expect: map[string]string{
+				rootID:         d.strPath(t),
+				folderID():     d.strPath(t, folderName()),
+				folderID("c1"): d.strPath(t, folderName(), folderName("c1")),
+				folderID("c2"): d.strPath(t, folderName(), folderName("c2")),
+			},
+		},
+		{
+			// tests the equivalent of:
+			// mv    root:/foo         -> root:/bar
+			// mkdir root:/foo
+			// mkdir root:/foo/c1
+			// mv    root:/bar/c1/c2   -> root:/foo/c1/c2
+			name: "moved and replaced with same name",
+			collectables: map[string]collectable{
+				rootID: {
+					prevPath: d.fullPath(t),
+					currPath: d.fullPath(t),
+				},
+				folderID(): {
+					prevPath: d.fullPath(t, folderName("foo")),
+					currPath: d.fullPath(t, folderName("bar")),
+				},
+				folderID(2): {
+					currPath: d.fullPath(t, folderName("foo")),
+				},
+				folderID("2c1"): {
+					currPath: d.fullPath(t, folderName("foo"), folderName("c1")),
+				},
+				folderID("c2"): {
+					prevPath: d.fullPath(t, folderName("bar"), folderName("c1"), folderName("c2")),
```

Comment:
I think actually this previous path should still be `root/foo/c1/c2` since this is all done by folder name not ID?

Reply:
right!  right.